### PR TITLE
DT-283: Add in java agent and remove w3c interceptor

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/config/RestTemplateConfiguration.java [hunk 1]
```diff
@@ -57,31 +54,23 @@ public RestTemplate elite2ApiHealthRestTemplate(final RestTemplateBuilder restTe
...
-                .additionalInterceptors(getRequestInterceptors())
+                .additionalInterceptors(new JwtAuthInterceptor())
                 .build();
     }
 
     private RestTemplate getHealthRestTemplate(final RestTemplateBuilder restTemplateBuilder, final String uri) {
         return restTemplateBuilder
                 .rootUri(uri)
-                .additionalInterceptors(getRequestInterceptors())
+                .additionalInterceptors(new JwtAuthInterceptor())
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/keyworker/config/RestTemplateConfiguration.java]: The request interceptor is replaced, what does this change mean?

Comment (petergphillips) [reply src/main/java/uk/gov/justice/digital/hmpps/keyworker/config/RestTemplateConfiguration.java]: the method takes either a list or varargs, only need the JWT interceptor here as the W3C tracing one happens automatically with the java agent

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/config/RestTemplateConfiguration.java [hunk 2]
```diff
@@ -57,31 +54,23 @@ public RestTemplate elite2ApiHealthRestTemplate(final RestTemplateBuilder restTe
...
                 .setConnectTimeout(healthTimeout)
                 .setReadTimeout(healthTimeout)
                 .build();
     }
 
-    private List<ClientHttpRequestInterceptor> getRequestInterceptors() {
-        return List.of(
-                new W3cTracingInterceptor(),
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/keyworker/config/RestTemplateConfiguration.java]: don't need the w3c tracing interceptor here

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/config/RestTemplateConfiguration.java (hunk 1 above)
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/digital/hmpps/keyworker/config/RestTemplateConfiguration.java]: cool

**Pull request merged**